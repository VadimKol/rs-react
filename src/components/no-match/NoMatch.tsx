import { type ReactNode } from 'react';
import { useNavigate } from 'react-router-dom';

import error_img from '@/assets/images/404.png';

import { CustomButton } from '../custom-button/Ð¡ustomButton';
import styles from './styles.module.scss';

export function NoMatch(): ReactNode {
  const navigate = useNavigate();
  const theme = 'dark';

  return (
    <main className={theme === 'dark' ? 'main' : 'main light'}>
      <img className={styles.error_img} src={error_img} alt="404 error" width={490} height={490} />
      <div className={styles.buttons}>
        <CustomButton className={styles.home} onClick={() => navigate('/')}>
          Home page
        </CustomButton>
      </div>
    </main>
  );
}
