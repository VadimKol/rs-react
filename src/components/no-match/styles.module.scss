.buttons {
  display: flex;
  gap: 30px;
  justify-content: space-between;
}

.error_img {
  display: block;

  @media screen and (width <= 550px) {
    width: 100%;
    height: 100%;
  }
}

.home {
  font-size: 1.3rem;
}
