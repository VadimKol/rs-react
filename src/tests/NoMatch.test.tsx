import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { NoMatch } from '@/components/no-match/NoMatch';

const navigate = vi.fn((arg: string): string => arg);

describe('NoMatch or 404 page', () => {
  const user = userEvent.setup();

  it('renders correctly', () => {
    vi.mock('@remix-run/react', () => ({ useNavigate: vi.fn(() => navigate) }));

    const { container } = render(<NoMatch />);

    expect(container).toMatchSnapshot();
  });

  it('navigating to Home when click on button', async () => {
    vi.mock('@remix-run/react', () => ({ useNavigate: vi.fn(() => navigate) }));

    render(<NoMatch />);

    const button = screen.getByRole('button');

    await user.click(button);

    expect(navigate).toHaveBeenCalledWith(`/`);
  });
});
