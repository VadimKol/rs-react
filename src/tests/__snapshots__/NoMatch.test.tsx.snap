// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoMatch or 404 page > renders correctly 1`] = `
<div>
  <main
    class="main"
  >
    <img
      alt="404 error"
      class="_error_img_788af9"
      height="490"
      src="/src/assets/images/404.png"
      width="490"
    />
    <div
      class="_buttons_788af9"
    >
      <button
        aria-label="Button"
        class="_button_de8d5c _home_788af9"
        id=""
        type="button"
      >
        Home page
      </button>
    </div>
  </main>
</div>
`;
